scipy/ndimage/morphology.py

```diff
@@ -240,6 +240,8 @@ def _binary_erosion(input, structure, iterations, mask, output,
         output = bool
     output = _ni_support._get_output(output, input)
 
+    if not isinstance(iterations, int):
+        raise RuntimeError('must run for an integer number of iterations')
```

Comment:
The recommended idiom is to use `operator.index(iterations)`, which will raise a TypeError for floats.
Also needs a test for np.int32, int64.

Reply:
Thanks for the feedback, @ev-br! I've updated this to use `operator.index`.

I've also added tests to verify that int64 and int32 are valid datatypes, which, as you mention, wouldn't have passed here.